src/github-cli/install.sh

```diff
@@ -124,138 +124,85 @@ check_packages() {
+    for version; do
+        download_url="https://github.com/cli/cli/releases/download/v${version}/gh_${version}_linux_${arch}.deb"
+        wget -P /tmp/ghcli "$download_url" || continue
+        dpkg -i "/tmp/ghcli/$(basename "$download_url")"
+        break
+    done
     rm -rf /tmp/ghcli
 }
 
-export DEBIAN_FRONTEND=noninteractive
-
-# Install curl, apt-transport-https, curl, gpg, or dirmngr, git if missing
-check_packages curl ca-certificates apt-transport-https dirmngr gnupg2
-if ! type git > /dev/null 2>&1; then
-    check_packages git
+if [ -n "$TEST_EXPORT" ]; then
+    # print function named by TEST_EXPORT and exit
+    declare -f "$TEST_EXPORT"
+    exit 0
 fi
 
-# Soft version matching
-if [ "${CLI_VERSION}" != "latest" ] && [ "${CLI_VERSION}" != "lts" ] && [ "${CLI_VERSION}" != "stable" ]; then
-    find_version_from_git_tags CLI_VERSION "https://github.com/cli/cli"
-    version_suffix="=${CLI_VERSION}"
-else
-    version_suffix=""
+check_packages ca-certificates
+type -P git >/dev/null || check_packages git
+
+is_latest="false"
+
+case "$CLI_VERSION" in
+none )
+    # opt out of installing gh
+    exit 0
+    ;;
+latest | lts | stable | current )
+    # find two latest versions
+    versions=( $(find_version_from_git_tags | head -n2) )
+    is_latest="true"
+    ;;
+*.*.* )
+    # the exact version is specified explicitly
+    versions=( "$CLI_VERSION" )
+    ;;
+* )
+    # find two latest versions matching version prefix
+    versions=( $(find_version_from_git_tags "$CLI_VERSION" | head -n2) )
+    ;;
+esac
+
+if [ "${#versions[@]}" -eq 0 ]; then
+    printf 'error: no git tags in the cli/cli repository matched version "%s"\n' "$CLI_VERSION" >&2
+    exit 1
 fi
 
 # Install the GitHub CLI
 echo "Downloading github CLI..."
 
 if [ "${INSTALL_DIRECTLY_FROM_GITHUB_RELEASE}" = "true" ]; then
-    install_deb_using_github
+    check_packages wget
+    install_deb_using_github "${versions[@]}"
+elif [ "$is_latest" != "true" ]; then
+    # In its current implementation, the https://cli.github.com/packages repository does not retain older versions of gh,
+    # so we cannot allow installing arbitrary versions with the `apt-get install gh=VERSION` syntax.
+    printf 'error: cannot install non-latest version when "installDirectlyFromGitHubRelease" is disabled\n' >&2
+    exit 1
 else
-   # Import key safely (new method rather than deprecated apt-key approach) and install
+    check_packages curl apt-transport-https dirmngr gnupg2
+    # Import key safely (new method rather than deprecated apt-key approach) and install
     . /etc/os-release
     receive_gpg_keys GITHUB_CLI_ARCHIVE_GPG_KEY /usr/share/keyrings/githubcli-archive-keyring.gpg
     echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" > /etc/apt/sources.list.d/github-cli.list
     apt-get update
-    apt-get -y install "gh${version_suffix}"
+    apt-get -y install gh
```

Comment:
Good to know. Thanks! 

Can we also add a description to [installDirectlyFromGitHubRelease property](https://github.com/devcontainers/features/blob/e7ebc0e9de1e3cf61424096352561d94b7e3e9a0/src/github-cli/devcontainer-feature.json#L17) to prompt that?

Something like 👇 

> If false, it installs the latest version of GitHub CLI